src/components/address-editor/address-editor.ts

```diff
@@ -0,0 +1,302 @@
+import Component from 'vue-class-component';
+import { Emit, Model, Prop, Watch } from 'vue-property-decorator';
+import Address, { AddressField, copyAddress, Country, CountryKey, Province } from '../../utils/address-lookup/address';
+import { ModulVue } from '../../utils/vue/vue';
+import { MDropdown } from '../dropdown/dropdown';
+import { MTextfield } from '../textfield/textfield';
+import WithRender from './address-editor.html';
+
+export type AddressEditorValidator = (value?: string, context?: Address, provinces?: Province[]) => string;
+
+@WithRender
+@Component({
+    components: {
+        MTextfield,
+        MDropdown
+    }
+})
+export default class MAddressEditor extends ModulVue {
+    @Model('change')
+    readonly address: Address;
+
+    @Prop({
+        required: true,
+        validator: (value: any) => value !== undefined
+    })
+    countries: Country[];
+
+    @Prop({
+        default: () => ([])
+    })
+    provinces: Province[];
+
+    @Prop({ default: () => ({}) })
+    validations: { [field: string]: AddressEditorValidator[] };
+
+    i18nBuildingNumber: string = this.$i18n.translate('m-address-editor:building-number');
+    i18nSubBuilding: string = this.$i18n.translate('m-address-editor:sub-building');
+    i18nStreet: string = this.$i18n.translate('m-address-editor:street');
+    i18nCity: string = this.$i18n.translate('m-address-editor:city');
+    i18nCountry: string = this.$i18n.translate('m-address-editor:country');
+    i18nProvince: string = this.$i18n.translate('m-address-editor:province');
+    i18nPostalCode: string = this.$i18n.translate('m-address-editor:postal-code');
+
+
+    currentAddress: Address = copyAddress(this.address);
+    private errors: { [field: string]: string[] } = {
+        [AddressField.BUILDING_NUMBER]: [],
+        [AddressField.STREET]: [],
+        [AddressField.SUB_BUILDING]: [],
+        [AddressField.CITY]: [],
+        [AddressField.PROVINCE]: [],
+        [AddressField.COUNTRY]: [],
+        [AddressField.POSTAL_CODE]: []
+    };
+
+    private touched: { [field: string]: boolean } = {
+        [AddressField.BUILDING_NUMBER]: false,
+        [AddressField.STREET]: false,
+        [AddressField.SUB_BUILDING]: false,
+        [AddressField.CITY]: false,
+        [AddressField.PROVINCE]: false,
+        [AddressField.COUNTRY]: false,
+        [AddressField.POSTAL_CODE]: false
+    };
+
+    mounted(): void {
+        this.isValid();
+    }
+
+    @Watch('address')
+    onAddressChange(): void {
+        this.currentAddress = this.address;
+        this.updateCountryCode(this.currentAddress[AddressField.COUNTRY][CountryKey.COUNTRY_ISO2]);
+    }
+
+    @Watch('province')
+    onProvincesChange(): void {
+        this.validate(AddressField.PROVINCE);
```

Comment:
`
Vu que provinces dépend directement de currentAddress
get province(): string {
        return (this.currentAddress.province) ? this.currentAddress.province.provinceCode : '';
    }

`

tu pourrais mettre this.validate(AddressField.PROVINCE); directement dans le setter plus haut normalement :)
if (address.province !== this.internalCurrentAddress.province) => validate province